File: Python/lkdngolf.py
# cook your dish here

T = int(input())

for i in range(T):
    
    inputs = input().split(' ')
    
    N, x, k = list(map(lambda n: int(n), inputs))

    tc = []
    
    flag = False
    
    for j in range(0, N+2, k):
        if (j > x):
            break
        tc.append(j)
    
    for j in range(len(tc)):
        if tc[j] == x:
            flag = True
            break
    
    # print(tc)
    tc = []
    
    for j in range(N+1, 0, -k):
        if (j < x):
            break
        tc.append(j)
    
    for j in range(len(tc)):
        if tc[j] == x:
            flag = True
            break
    
    # print(tc)

    if(flag):
        print("Yes")
    else:
        print("No")

File: Java/lkdngolf.java
/* package codechef; // don't place package name! */

import java.util.*;
import java.lang.*;
import java.io.*;

/* Name of the class has to be "Main" only if the class is public. */
class Codechef
{
	public static void main (String[] args) throws java.lang.Exception
	{
		// your code goes here
		Scanner sc = new Scanner(System.in);

        int T = sc.nextInt();
        
        for(int i = 0; i<T; i++) {
            int N = sc.nextInt();
            int x = sc.nextInt();
            int k = sc.nextInt();
            
            ArrayList<Integer> tc = new ArrayList<Integer>();
            
            for (int j = N+1; j >= 0; j-=k) {
                    if (j < x)
                        break;
                    tc.add(j);
                }
            
            boolean flag = false;
            
            for (int j = 0; j < tc.size(); j++)
                if (tc.get(j) == x) {
                    flag = true;
                    break;
                }
        
            if((x % k) == 0 || flag)
                System.out.println("Yes");
            else
                System.out.println("No");
        }
	}
}
